File: ui/src/app/edge/installation/views/configuration-execute/safety-country.ts
import { Country } from "src/app/shared/type/country";

export enum SafetyCountry {
    GERMANY = 'GERMANY',
    AUSTRIA = 'AUSTRIA',
    SWITZERLAND = 'SWITZERLAND',
    SWEDEN = 'SWEDEN',
    CZECH_REPUBLIK = 'CZECH',
    NETHERLANDS = 'HOLLAND',
}

export namespace SafetyCountry {
    export function getSafetyCountry(countryValue: Country) {
        switch (countryValue) {
            case Country.GERMANY:
                return SafetyCountry.GERMANY;
            case Country.AUSTRIA:
                return SafetyCountry.AUSTRIA;
            case Country.SWITZERLAND:
                return SafetyCountry.SWITZERLAND;
            case Country.CZECH_REPUBLIK:
                return SafetyCountry.CZECH_REPUBLIK;
            case Country.NETHERLANDS:
                return SafetyCountry.NETHERLANDS;
            case Country.SWEDEN:
                return SafetyCountry.SWEDEN;
        }
    }
}

File: io.openems.edge.core/src/io/openems/edge/app/enums/SafetyCountry.java
package io.openems.edge.app.enums;

import io.openems.common.session.Language;
import io.openems.edge.core.appmanager.AbstractOpenemsApp;
import io.openems.edge.core.appmanager.TranslationUtil;

public enum SafetyCountry implements TranslatableEnum {
	GERMANY("germany"), //
	AUSTRIA("austria"), //
	SWITZERLAND("switzerland"), //
	SWEDEN("sweden"), //
	CZECH("czech"), //
	HOLLAND("netherlands"), //
	IRELAND("ireland"), //
	UNITED_KINGDOM("united_kingdom"), //
	;

	private final String translationKey;

	private SafetyCountry(String translationKey) {
		this.translationKey = translationKey;
	}

	@Override
	public final String getTranslation(Language l) {
		final var bundle = AbstractOpenemsApp.getTranslationBundle(l);
		return TranslationUtil.getTranslation(bundle, this.translationKey);
	}

	/**
	 * Creates a {@link OptionsFactory} of this enum.
	 * 
	 * @return the {@link OptionsFactory}
	 */
	public static final OptionsFactory optionsFactory() {
		return OptionsFactory.of(values());
	}

}
